runtime/drivers/druid/druidsqldriver/druid_api_sql_driver.go

```diff
@@ -40,193 +56,223 @@ type sqlConnection struct {
 
 var _ driver.QueryerContext = &sqlConnection{}
 
-func (c *sqlConnection) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
-	dr := newDruidRequest(query, args)
-	b, err := json.Marshal(dr)
-	if err != nil {
-		return nil, err
-	}
+func (c *sqlConnection) Prepare(query string) (driver.Stmt, error) {
+	return &stmt{
+		query: query,
+		conn:  c,
+	}, nil
+}
 
-	bodyReader := bytes.NewReader(b)
+func (c *sqlConnection) Close() error {
+	return nil
+}
 
-	context.AfterFunc(ctx, func() {
-		tctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-		defer cancel()
-		r, err := http.NewRequestWithContext(tctx, http.MethodDelete, c.dsn+"/"+dr.Context.SQLQueryID, http.NoBody)
-		if err != nil {
-			return
-		}
+func (c *sqlConnection) Begin() (driver.Tx, error) {
+	return nil, fmt.Errorf("unsupported")
+}
 
-		resp, err := c.client.Do(r)
+func (c *sqlConnection) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
+	// total sum is 126 seconds (sum(2*2^x) from 0 to 5 inclusive)
+	re := retrier.NewRetrier(6, 2*time.Second, &coordinatorHTTPCheck{
+		c: c,
+	})
+	return re.RunCtx(ctx, func(ctx2 context.Context) (driver.Rows, retrier.Action, error) {
```

Comment:
Is it on purpose that most calls use `ctx` instead of `ctx2`? What is the implication of using one vs. the other?

Reply:
Removed.